libraries/xcode.rb

```diff
@@ -3,18 +3,20 @@
 
 module MacOS
   class Xcode
-    attr_reader :version
-    attr_reader :intended_path
-    attr_reader :download_url
+    attr_reader :version, :intended_path, :download_url
 
     def initialize(semantic_version, intended_path, download_url = '')
       @semantic_version = semantic_version
       @intended_path = intended_path
       @download_url = download_url
-      @version = if download_url.empty?
-                   latest_xcode_revision(Xcode::Version.new(semantic_version))
+      @version = -> { determine_version }.call
```

Comment:
Possible alternative to the lamda:

`@version = download_url.empty? ? latest_xcode_revision(Xcode::Version.new(@semantic_version)) : @samantic_version`

Reply:
The lambda allows the `xcode` object to be instantiated without authenticating with Apple to see what versions are available, and allows us to first check things like `xcode.installed?` for the resource guards. The ternary would still be evaluated at initialization, which will require auth with Apple right away.